### DISCUSSION TITLE
Set particle effects for blocks

### DISCUSSION THREAD

**Discussion by alazeprt** [FabricMC/fabric-api#2673] [Mod Dev Support]
How to set particle effects for blocks?

**Comment by warjort** [comment]
You need to make the block tickable and spawn the particles yourself.
You can make a block tickable by making it a BlockEntity - see CampfireBlockEntity.clientTick()

Another way would be to override hasRandomTicks() and randomTick() like crop growth.
But that doesn't sound like the right solution for what you are doing since it is sporadic and runs on the server.
